docs/dev/design/new-search-api.rst

```diff
@@ -0,0 +1,238 @@
+New search API
+==============
+
+Goals
+-----
+
+- Allow to configure search at the API level,
+  instead of having the options in the database.
+- Allow to search a group of projects/versions at the same time.
+
+Syntax
+------
+
+The parameters will be given in the query using the ``key:value`` syntax.
+Inspired by `GitHub <https://docs.github.com/en/rest/search>`__ and other services.
+
+Currently the values from all parameters don't include spaces,
+so surrounding the value with quotes won't be supported (``key:"value"``).
+
+To avoid interpreting a query as a parameter,
+an escape character can be put in place,
+for example ``project\:docs`` won't be interpreted as
+a parameter, but as the search term ``project:docs``.
+This is only necessary if the query includes a valid parameter,
+unknown parameters (``foo:bar``) don't require escaping.
+
+All other tokens that don't match a valid parameter,
+will be join to form the final search term.
+
+Parameters
+----------
+
+project:
+ Indicates the project and version
+ to includes results from (this doesn't include subprojects).
+ If the version isn't provided,
+ the default version is used.
+
+ Examples:
+
+ - ``project:docs/latest``
+ - ``project:docs``
```

Comment:
Hrm, I wonder if we want `version` as an optional second argument?

Reply:
Do you mean having `version` as another parameter? The version is given together with the project, this is so we can associate a version to each project, instead of having one version for all projects, if the version isn't given we use the default version.

Reply:
I understand why you decided to omit the `version` attribute and it makes sense to me.

The only thing that makes me a little of noise is the name of the attribute, `project`, that communicate nothing about the ability to add the version as well. I don't have a good suggestion, tho.